Fixing Errors (Update Connexion Library to version 3)

Satoshi-Sh opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->



<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->
## Description

This PR was created based on  #36052
## Todo
- [x] Fix the issue that  `auth_mgr.get_api_endpoints(connexion_app)` returns None instead of a blueprint object 
- [ ] Find a way to replace Legacy environ_overrides method from old version of Connexion to fix the problem in tests.
- [ ] Fix unit tests that use app

<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

Commit:
Update methods to use Connexion v3, Ginucorn command and encoding

Commit:
Fix static checks

MaksYermak committed:
Update setup.cfg and encoding

MaksYermak committed:
Update migrations files for

MaksYermak committed:
Update configuration for tests

MaksYermak committed:
Fix problem with static checks

Satoshi-Sh committed:
Followed vincbeck instruction on the issue #36052

vincbeck commented on airflow/cli/commands/internal_api_command.py:
```diff
@@ -240,11 +241,11 @@ def create_app(config=None, testing=False):
         init_jinja_globals(flask_app)
         init_xframe_protection(flask_app)
-    return flask_app
+    return connexion_app
```
Why not returning the `flask_app` instead of the `connexion_app`? That would avoid making many changes around this

sudiptob2 replied:
Thanks, got it. here `connexion_app.app` returns a `FlaskApp` so we can return it. Can you explain or give some reference on the benefits you anticipate? We do not have the full context of this PR yet. Will help to understand the migration better 💯

vincbeck replied:
The will decrease the number of changes. There are many changes in this PR which replace `cached_app(config={"FAB_UPDATE_PERMS": False}).appbuilder` by `cached_app(config={"FAB_UPDATE_PERMS": False}).app.appbuilder`. By doing so, we will no longer need these changes

sudiptob2 replied:
Thanks! Got it 🚀

vincbeck commented on airflow/cli/commands/internal_api_command.py:
```diff
@@ -73,8 +73,8 @@ def internal_api(args):
         app.run(
-            debug=True,  # nosec
-            use_reloader=not app.config["TESTING"],
+            log_level="debug",
+            # reload=not app.app.config["TESTING"],
```
What's the change here?

vincbeck commented on airflow/providers/fab/auth_manager/fab_auth_manager.py:
```diff
@@ -148,21 +150,26 @@ def get_cli_commands() -> list[CLICommand]:
             specification = safe_load(f)
-        return FlaskApi(
+
+        swagger_ui_options = SwaggerUIOptions(
+            swagger_ui=conf.getboolean("webserver", "enable_swagger_ui", fallback=True),
+        )
```
I think this come from another PR? If so please rebase to make the review easier

vincbeck commented on airflow/providers/fab/auth_manager/fab_auth_manager.py:
```diff
@@ -148,21 +150,26 @@ def get_cli_commands() -> list[CLICommand]:
+        )
+
+        connexion_app.extensions["csrf"].exempt(api.blueprint)
+        return api.blueprint if api else None
```
In which case api can be Falsy?

sunank200 committed:
Refactor dataset class inheritance (#37590)

* Refactor DatasetAll and DatasetAny inheritance

They are moved from airflow.models.datasets to airflow.datasets since
the intention is to use them with Dataset, not DatasetModel. It is more
natural for users to import from the latter module instead.

A new (abstract) base class is added for the two classes, plus the OG
Dataset class, to inherit from. This allows us to replace a few
isinstance checks with simple molymorphism and make the logic a bit
simpler.

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>
Co-authored-by: Wei Lee <weilee.rx@gmail.com>

potiuk committed:
Check if Python 3.12 is used for release management commands as well (#37615)

We've been checking - for reproducibility - if python version used is
Python 3.9 or above, but since we are also rebuilding sdist packages,
we need to check if sdist packages can be converted to wheel packages
as well so we need to check python version used.

amoghrajesh committed:
Enhancing breeze commands with PACKAGE_LIST env variable (#37502)

* Enhancing breeze commands with PACKAGE_LIST env variable

* removing from add-back-references as it is idempotent

* review comments from potiuk

* fixing doc building

* fixing static checks

Owen-CH-Leung committed:
Introduce new config variable to control whether DAG processor outputs to stdout (#37439)

hterik committed:
Clarify skip-message for test requiring database. (#37612)

The error was misleading by saying envvar should be 'true',
instead of the expected value.

hterik committed:
Use new exception type inheriting BaseException for SIGTERMs (#37613)

When a task is terminated, it get sent a SIGTERM signal,
(through standard_task_runner.terminate).
This signal is picked up by taskinstance.py, where it injects an
AirflowException into the current running task.

The exception raised there is now changed into a new
AirflowTaskTerminated exception.
This new exception inherits BaseException, so that it
is not caught and ignored accidentally by code that tries
to perform ordinary error handling. Because termination requests
should not be ignored.

This will improve reliability of cancelling tasks and will make
error logs and error categorization more clear about the reason
for failure.

(The terminate function does eventually send SIGKILL if the
 SIGTERM is ignored, but waiting for that should not be necessary)

See https://docs.python.org/3/library/exceptions.html#exception-hierarchy

vincbeck committed:
Implement `batch_is_authorized_*` APIs in AWS auth manager (#37430)

kacpermuda committed:
chore: Update comments and logging in OpenLineage ExtractorManager (#37622)

sudiptob2 committed:
Merge branch 'main' into fix/api-endpoints-registration

sudiptob2 commented on airflow/providers/fab/auth_manager/fab_auth_manager.py:
```diff
@@ -149,19 +150,26 @@ def get_cli_commands() -> list[CLICommand]:
-        ).blueprint
+        )
+
+        connexion_app.extensions["csrf"].exempt(api.blueprint)
```
`'FlaskApp' object has no attribute 'extensions'`, So I think it is not possible to do it like this.

sudiptob2 committed:
fix: add connexion v3 and missing dependencies in pyproject.toml.

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
fix: use appbuilder.app instead of FlaskApp

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

bbovenzi committed:
Show dataset events above task/run details in grid view (#37603)

* Show dataset events above task/run details in grid view

* Add extra to dataset event tables

potiuk committed:
Update min kubernetes version in docs and update release instructions (#37623)

The version in docs has been out-of-sync with the README. This PR
fixes it and adds instruction in release process to keep them in
sync.

eladkal committed:
Prepare docs ad hoc Teradata provider RC2 February 2024 (#37624)

* Prepare docs ad hoc Teradata provider RC2 February 2024

* Update change log

bbovenzi committed:
Add datasets to dag graph (#37604)

* Add dataset nodes to dag graph

* Clean up dataset nodes

bbovenzi committed:
Filter datasets graph by dag_id (#37464)

* Initial filter datasets graph by dag_id

* Fix bug with merging dataset subgraphs

* Fix index for dataset group merging

* Add tooltip

bbovenzi committed:
Fix task duration selection (#37630)

pankajastro committed:
Fix Async GCSObjectsWithPrefixExistenceSensor xcom push (#37634)

Fix GCSObjectsWithPrefixExistenceSensor found prefix in fist
poke only in async mode of sensor then it does not push matches
in xcom. This PR fix it.

potiuk committed:
Fix few small release issues found during 2.8.2 preparation (#37633)

* when deleting the whole .build folder, the www folder could not be
  created during asset compilation
* -source packages were treated as sdist and we attempted to build
  wheel files with them
* when building airflow packages, default settting (for security
  reasons) is to build everything in docker container, but building
  wheel for sdist was done outside. With this PR we only use sdist
  check when we use local hatch build - and sdist wheel check
  happens in docker container when sdists are built.
* tarball was prepared before the pypi packages and it has been
  deleted by local hatch build

sudiptob2 committed:
feat: return connexion.app to keep the return type as FlaskApp

Signed-off-by: sudipto baral <sudiptobaral.me@gmail.com>

sudiptob2 committed:
Merge branch 'main' into fix/api-endpoints-registration
